File: benchmarks/python/run_all.py
"""Run all Python benchmark scripts sequentially."""

import subprocess
import sys
from pathlib import Path

scripts = sorted(Path("benchmarks/python").glob("[0-9][0-9]_*.py"))

print("=" * 40)
print("Running all Python benchmarks")
print("=" * 40)

for script in scripts:
    print(f"\n--- Running: {script} ---")
    result = subprocess.run([sys.executable, str(script)], capture_output=False)
    if result.returncode != 0:
        print(f"ERROR: {script} failed with exit code {result.returncode}")

print("\n" + "=" * 40)
print("All Python benchmarks complete.")
print("Results in: benchmarks/python/results/")
print("=" * 40)

File: benchmarks/r/run_all.R
# Run all R benchmark scripts sequentially
cat("========================================\n")
cat("Running all R benchmarks\n")
cat("========================================\n\n")

scripts <- list.files("benchmarks/r", pattern = "^\\d{2}_.*\\.R$", full.names = TRUE)
scripts <- sort(scripts)

for (s in scripts) {
  cat("\n--- Running:", s, "---\n")
  tryCatch(
    source(s, local = new.env()),
    error = function(e) cat("ERROR:", conditionMessage(e), "\n")
  )
}

cat("\n========================================\n")
cat("All R benchmarks complete.\n")
cat("Results in: benchmarks/r/results/\n")
cat("========================================\n")